src/target/riscv/program.c

```diff
@@ -100,236 +79,39 @@ int riscv_program_exec(struct riscv_program *p, struct target *t)
 	return ERROR_OK;
 }
 
-riscv_addr_t riscv_program_alloc_data(struct riscv_program *p, size_t bytes)
```

Comment:
It's not necessary. It can only be used when the program buffer is large enough to accommodate this kind of stuff. Since the 2-word buffer code supports all features, there's no need to support doing things a different way that does not provide a benefit.

Reply:
"supports all features" is true at the moment for the features we have decided to implement, namely GPR, CSR, and memory access. But the whole point of the program buffer is that it can support other features, like vector register access. Would you expect to just add this functionality back in once it becomes needed?

Reply:
Yes. I don't think it's helpful to keep around a bunch of code just in case it will be needed some day. If we'll be using it imminently that's different, but I cannot think of anything I want to implement that would use it.